File: application-bemyplan/src/main/java/com/deploy/bemyplan/controller/auth/dto/request/LoginRequestDto.java
package com.deploy.bemyplan.controller.auth.dto.request;

import com.deploy.bemyplan.domain.user.UserSocialType;
import com.deploy.bemyplan.service.auth.dto.request.LoginDto;
import lombok.*;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

@ToString
@Getter
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class LoginRequestDto {

    @NotBlank(message = "{auth.token.notBlank}")
    private String token;

    @NotNull(message = "{user.socialType.notNull}")
    private UserSocialType socialType;

    public LoginDto toServiceDto() {
        return LoginDto.of(token, socialType);
    }
}

File: applications/app-bemyplan/src/main/kotlin/com/bemyplan/auth/api/LoginRequestDto.kt
package com.bemyplan.auth.api

import com.deploy.bemyplan.auth.service.dto.LoginDto
import com.deploy.bemyplan.domain.user.UserSocialType
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

data class LoginRequestDto(
    @NotBlank(message = "{auth.token.notBlank}") val token: String,
    @NotNull(message = "{user.socialType.notNull}") val socialType: UserSocialType,
) {
    fun toServiceDto(): LoginDto {
        return LoginDto.of(token, socialType)
    }
}
